File: Website_Phim/WebPhim-Backend/MovieWebsite/MovieWebsite/src/main/java/com/project/MovieWebsite/models/Movie.java
package com.project.MovieWebsite.models;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;


@Entity
@Table(name="tab_movie")
@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Movie {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name="name", nullable= false, length=255)
    private String name;

    @Column(name="description", nullable= false, length=255)
    private String description;

    @Column(name="image", nullable= false)
    private String image;

    private String slug;

    @Column(name="release_date", nullable= false)
    private Date releaseDate;

    @Column(name="duration", nullable= false, length=255)
    private String duration;

    @ManyToOne
    @JoinColumn(name="id_genre")
    private Genre genre;

    @ManyToOne
    @JoinColumn(name="id_movie_type")
    private MovieType movieType;

    @ManyToOne
    @JoinColumn(name="id_country")
    private Country country;

    @Column(name="episode")
    private int episode;

    @Column(name="hot", nullable= false)
    private int hot;

    @Column(name="is_fee", nullable= false)
    private int isFee;

    @Column(name="season", nullable= false)
    private int season;

    @Column(name="limited_age", nullable= false)
    private int limitedAge;

    @Column(name="number_view", nullable= false)
    private int numberView;

    @Column(name="is_active")
    private int isActive;

    @PrePersist
    protected void onCreate() {
        this.releaseDate = new Date();
    }

}

File: MovieWebsite-fontend/src/app/models/movie.ts

export interface Movie {
    id: number,
    name: string,
    description: string,
    image: string,
    slug: string,
    release_date: number,
    duration: string,
    id_genre: number,
    id_movie_type: number,
    id_country: number,
    episode: number,
    hot: number,
    is_fee: number,
    season: number,
    limited_age: number,
    number_views: number,
    movie_type_name: string,
    genre_name: string,
    country_name: string,
    url: string
    release_date_formated: string
    is_active: number
}